File: apps/spring-boot-api/src/main/java/org/bootstrapbugz/api/auth/event/listener/OnSendJwtEmailListener.java
package org.bootstrapbugz.api.auth.event.listener;

import org.bootstrapbugz.api.auth.event.OnSendJwtEmail;
import org.bootstrapbugz.api.auth.event.email.JwtEmailSupplier;
import org.bootstrapbugz.api.shared.email.service.EmailService;
import org.springframework.context.ApplicationListener;
import org.springframework.core.env.Environment;
import org.springframework.stereotype.Component;

@Component
public class OnSendJwtEmailListener implements ApplicationListener<OnSendJwtEmail> {
  private final EmailService emailService;
  private final Environment environment;

  public OnSendJwtEmailListener(EmailService emailService, Environment environment) {
    this.emailService = emailService;
    this.environment = environment;
  }

  @Override
  public void onApplicationEvent(OnSendJwtEmail event) {
    new JwtEmailSupplier()
        .supplyEmail(event.getPurpose())
        .sendEmail(emailService, environment, event.getUser(), event.getToken());
  }
}

File: backend/spring-boot-kotlin/src/main/kotlin/org/bootstrapbugz/api/auth/jwt/event/listener/OnSendJwtEmailListener.kt
package org.bootstrapbugz.api.auth.jwt.event.listener

import org.bootstrapbugz.api.auth.jwt.event.OnSendJwtEmail
import org.bootstrapbugz.api.auth.jwt.event.email.JwtEmailSupplier
import org.bootstrapbugz.api.shared.email.service.EmailService
import org.springframework.context.ApplicationListener
import org.springframework.core.env.Environment
import org.springframework.stereotype.Component

@Component
class OnSendJwtEmailListener(
    private val emailService: EmailService,
    private val environment: Environment
) : ApplicationListener<OnSendJwtEmail> {

    override fun onApplicationEvent(event: OnSendJwtEmail) {
        JwtEmailSupplier
            .supplyEmail(event.purpose)
            .sendEmail(emailService, environment, event.user, event.token)
    }
}